### PR TITLE
Avoid hook errors when effecting label changes on kubernetes-worker charm.

### PR DESCRIPTION
**Pull request by wwwtyro**
**What this PR does / why we need it**: Avoid hook errors when effecting label changes on kubernetes-worker charm.

**Release note**:
```release-note
Avoid hook errors when effecting label changes on kubernetes-worker charm.
```

### CODE REVIEW

File: cluster/juju/layers/kubernetes-worker/reactive/kubernetes_worker.py [hunk 1]
```diff
@@ -412,37 +412,55 @@ def render_and_launch_ingress():
...
-        # during the atomic update below.
-
-    # Atomically set a label
-    for label in user_labels:
-        _apply_node_label(label, overwrite=True)
-
-    # Set label for application name
-    _apply_node_label('juju-application={}'.format(hookenv.service_name()),
-                      overwrite=True)
+@when_any('kubernetes-worker.label-config-required')
```
Comment (ktsakalozos) [inline cluster/juju/layers/kubernetes-worker/reactive/kubernetes_worker.py]: There is something strange with this when_any

Comment (wwwtyro) [reply cluster/juju/layers/kubernetes-worker/reactive/kubernetes_worker.py]: Ah, thanks @ktsakalozos, nice catch.

**Pull request merged**